lib/_http_common.js

```diff
@@ -160,6 +161,7 @@ const parsers = new FreeList('parsers', 1000, function parsersCb() {
   parser[kOnHeadersComplete] = parserOnHeadersComplete;
   parser[kOnBody] = parserOnBody;
   parser[kOnMessageComplete] = parserOnMessageComplete;
+  parser[kOnMessageBegin] = null;
```

Comment:
my understanding was that onMessageBegin will be called when the parser starts parsing a new request's headers.

Reply:
why are you assigning it here instead of in http server?